### DISCUSSION TITLE
Create a periodic task with Celery

### DISCUSSION THREAD

**Discussion by yuleidyvdres** [saleor/saleor#12951] [General] [answered]
I need to create a periodic task but it's not recognized when building the container.

I created a plugin on `saleor/plugins/` called `message` and here's my `tasks.py` file, I just created this function to test 

```python
@shared_task
def dummy_function():
    logger.info('Running dummy function')
``` 

On `saleor/settings.py` I added 
```python
CELERY_BEAT_SCHEDULE = {
      ...,
      "dummy-function": {
        "task": "saleor.plugins.message.tasks.dummy_function",
        "schedule": crontab(minute="*/1")
    }
}
``` 

Then I run `docker compose up -d --build` and check the logs of the worker service, but the task is not listed 

![image](https://github.com/saleor/saleor/assets/26033728/f3204e13-4e7c-42e9-9c2f-27c7fb11df50)

Is there something missing? or what is the way to do it? 

I'm using saleor core v3.13.0-a

`CELERY_BROKER_URL = redis://redis:6379/1`

**Comment by korycins** [comment, marked as answer]
@yuleidyvdres For plugins, Saleor can discover the tasks defined in `tasks.py`. I just test it with your use case and it works. 
One thing that can be missing on your side:
Make sure that your plugin is added to `settings.py:PLUGINS`

Steps that I did:
- I've created a file: `saleor/payment/gateways/adyen/tasks.py` (`gateways/adyen` is a plugin)
- I ve added there a logic:
```
from celery import shared_task

@shared_task
def dummy_function():
    pass
```
- I've opened the `settings.py` file and checked that `PLUGINS` list contains `"saleor.payment.gateways.adyen.plugin.AdyenGatewayPlugin"`
- Then the new celery task is visible in  the worker's task list
